src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -24,13 +24,25 @@ public abstract partial class JsonSerializerContext : IJsonTypeInfoResolver
         /// </remarks>
         public JsonSerializerOptions Options
         {
-            get => _options ??= new JsonSerializerOptions { TypeInfoResolver = this, IsImmutable = true };
+            get
+            {
+                JsonSerializerOptions? options = _options;
+
+                if (options is null)
+                {
+                    options = new JsonSerializerOptions { TypeInfoResolver = this };
+                    options.MakeReadOnly();
+                    _options = options;
+                }
+
+                return options;
```

Comment:
I'm wondering if you should instead do something like:

```suggestion
                    _options ??= options;
                }

                return _options;
```

to ensure we return the same instance in the multi-threaded scenarios

Reply:
We can't really guarantee that the same instance gets returned in multi-threaded scenarios unless we use interlocked operations, but we don't really need to do that either.